main/solution/post-deployment/config/environment-files/clone_repos.sh

```diff
@@ -27,7 +29,9 @@ mkdir $repo_dir && cd $repo_dir
 
 for repo in ${repos[@]}; do
     echo $repo
-    GIT_TERMINAL_PROMPT=0 git clone $repo
+    if ! git clone -b $branch $repo; then
+        git clone $repo
+    fi
```

Comment:
I waffled around on how to do this a lot. Here are the iterations:
 - Change the array to `::` delimited tuples of repo-branch pairs, split during iteration
 - Use an associative array where the keys are repos and the values are branches

I felt that the split was bad because it's essentially parsing your own data, and splitting isn't exactly clean in bash
Associative arrays were a bit better, but I still felt that the syntax was gross. Also, they're a new enough feature that the script might not work in some environments.

This solution is incredibly dumb, but simple. It also makes the assumption that the branch will be either `$branch` or the repo default. This should be ok, since we control all the example repos.